guppylang/std/_internal/checker.py

```diff
@@ -179,34 +178,22 @@ class NonCopyableElementsError(Error):
         class Explanation(Note):
             message: ClassVar[str] = "Only arrays with copyable elements can be copied"
 
-    @dataclass(frozen=True)
-    class ExpectedArrayError(Error):
-        title: ClassVar[str] = "Non-array copy"
-        span_label: ClassVar[str] = (
-            "Expected array expression, got expression of type {ty}"
-        )
-        ty: Type
-
     def synthesize(self, args: list[ast.expr]) -> tuple[ast.expr, Type]:
-        check_num_args(1, len(args), self.node)
-        [array_arg] = args
-        array_arg, array_ty = ExprSynthesizer(self.ctx).synthesize(array_arg)
-        if not is_array_type(array_ty):
-            raise GuppyTypeError(
-                ArrayCopyChecker.ExpectedArrayError(self.node, array_ty)
-            )
```

Comment:
We shouldn't be hand-rolling our own checking logic here, just call `synthesize_call`. This makes sure we're doing all the required inout checks and also makes error messages more consistent.